substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -1084,6 +1084,35 @@ impl<T: Config> Pallet<T> {
 		outcome
 	}
 
+	/// TODO: docs
+	/// Basically another call for `bond`, but it's implied that the funds are already locked
+	/// and `pallet-staking` doesn't need to lock them here.
+	pub fn do_delegated_bond(
+		stash: T::AccountId,
+		value: BalanceOf<T>,
+		payee: RewardDestination<T::AccountId>,
+	) -> DispatchResult {
+		if StakingLedger::<T>::is_bonded(StakingAccount::Stash(stash.clone())) {
+			return Err(Error::<T>::AlreadyBonded.into())
+		}
+
+		// Reject a bond which is considered to be _dust_.
+		if value < T::Currency::minimum_balance() {
+			return Err(Error::<T>::InsufficientBond.into())
+		}
+
+		frame_system::Pallet::<T>::inc_consumers(&stash).map_err(|_| Error::<T>::BadState)?;
+		let stash_balance = T::Currency::free_balance(&stash);
```

Comment:
We can't look at free balance of the stash. We need to somehow query how much balance is delegated to this `stash` (delegatee)

Reply:
This check was here before for some other reason and now it's not even needed, so I removed it.